### PR TITLE
Fixing how trajectory files are written

### PR DESCRIPTION
**Pull request by bannanc**
Fixing how trajectory files are written out in smirky (and smarty). 
More details in issue #189

### CODE REVIEW

**Comment by bannanc** [comment]
@camizanette 
There is a test failing on the elemental sampler, which I certainly didn't make any changes to. The error message is below:
```
======================================================================
ERROR: test_atomtyper_elemental (smarty.tests.test_sampler.TestAtomTypeSampler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test/lib/python3.4/site-packages/smarty/tests/test_sampler.py", line 58, in test_atomtyper_elemental
    fracfound = atomtype_sampler.run(2)
  File "/home/travis/miniconda/envs/test/lib/python3.4/site-packages/smarty/sampler_elemental.py", line 906, in run
    roots.remove(child)
AttributeError: 'dict_keys' object has no attribute 'remove'
```

**Comment by bannanc** [comment]
Ok, I found what caused this error. I'm a little surprised this is the first time it has come up. However the remove that is causing the error doesn't always happen so if we're only doing 2 iterations in all the tests it might not come up. (heads up @camizanette )

In python2 when you call dict.keys() it returns a list of the keys in that dictionary. 
Instead in python3 it returns an iterator for the keys. 
[Here's a blog post](http://btmiller.com/2015/04/13/get-list-of-keys-from-dictionary-in-python-2-and-3.html) I found while trying to identify what was causing the error incase anyone wants to read more. 

Most of us in the Mobley group are still running python2.7 and were depending on the tests here to catch changes like this.

**Comment by bannanc** [comment]
I would like to include for the record, that the elemental sampler should have inherited almost all of its methods from the original sampler instead of copying code... I'm almost done making fixes to both now.

File: smarty/sampler.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def __init__(self, molecules, basetypes_filename, initialtypes_filename, decorat
         self.typetag = 'atomtype'
 
         # Save bond list to use throughout
-        self.bondset = [("-","simply"), ("=", "doubly"), ("#","triply"), (":", "aromatic"), ("~","any")]
+        self.bondset = [("-","singly"), ("=", "doubly"), ("#","triply"), (":", "aromatic"), ("~","any")]
```
Comment (bannanc) [inline smarty/sampler.py:90]: This typo has bothered me for a while and I alway forgot to fix it.

File: smarty/sampler.py [hunk 2]
```diff
@@ -731,6 +743,21 @@ def get_unfinishedAtomList(self, atom_typecounts, molecule_typecounts, atomtype_
 
         return
 
+    def _switch_parent_dict(self):
```
Comment (bannanc) [inline smarty/sampler.py:746]: I wanted to be able to get the parent if I knew the child. This is the best I could come up with without completely restructuring the parent dictionary. 
In smirky the parent dictionary is an embedded dictionary so it stores children and parent for every parameter type. I could update smarty to do that instead if we want consistency

File: smarty/sampler_elemental.py [hunk 3]
```diff
@@ -88,13 +88,13 @@ def __init__(self, molecules, basetypes_filename, initialtypes_filename, decorat
         This is just a proof of concept.  No scoring of molecular properties is performed.
 
         """
-        
-        
+
+
```
Comment (bannanc) [inline smarty/sampler_elemental.py:92]: I have my vim settings delete extra spaces at the end of a line. 
I'm assuming whoever edited this last did not have those settings in place...

Comment (davidlmobley) [reply smarty/sampler_elemental.py:92]: Hmm, that's odd. Everyone who touched it was told to set that, but perhaps someone missed it.

**Review by vtlim — APPROVED** [review]
Looks fine to me. (although a bit harder to evaluate out of context)

**Pull request merged**